azure-devops/azext_devops/dev/team/security_permission.py

```diff
@@ -0,0 +1,212 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import json
+from knack.util import CLIError
+from knack.log import get_logger
+from azext_devops.devops_sdk.exceptions import AzureDevOpsClientRequestError
+from azext_devops.devops_sdk.v5_0.security.models import (AccessControlEntry)
+from azext_devops.dev.common.services import (get_security_client,
+                                              get_graph_client,
+                                              resolve_instance)
+from azext_devops.dev.common.identities import (get_identity_descriptor_from_subject_descriptor,
+                                                resolve_identity_as_identity_descriptor)
+from azext_devops.dev.common.uuid import is_uuid
+from .security_permission_helper import PermissionDetails
+logger = get_logger(__name__)
+
+
+def list_namespaces(local_only=False, organization=None, detect=None):
+    """ List all available namespaces for an organization.
+    :param local_only: If true, retrieve only local security namespaces.
+    :type local_only: bool
+    """
+    organization = resolve_instance(detect=detect, organization=organization)
+    client = get_security_client(organization)
+    response = client.query_security_namespaces(local_only=local_only)
+    return response
+
+
+def show_namespace(namespace_id, organization=None, detect=None):
+    """ Show details of permissions avaialble in each namespace.
+    """
+    organization = resolve_instance(detect=detect, organization=organization)
+    client = get_security_client(organization)
+    response = _get_permission_types(client, namespace_id)
+    return response
+
+def list_tokens(namespace_id, subject, token=None,
+                recurse=False, organization=None, detect=None):
+    """ List tokens for given user/group and namespace.
+    :param recurse: If true and this is a hierarchical namespace, return child ACLs of the specified token.
+    :type recurse: bool
+    """
+    organization = resolve_instance(detect=detect, organization=organization)
+    client = get_security_client(organization)
+    if '@' in subject:
+        subject = resolve_identity_as_identity_descriptor(subject, organization)
+    elif not is_uuid(subject) and '.' in subject:
```

Comment:
nit: '.' in subject will always be NOT uuid, so checking for '.' in subject should be sufficient?